Restructured app

vorugantia opened the pull request:
Pull request contains: 
- Most recent dependencies in package.json
- Fully refactored and restructured app
- Just a (working) skeleton of the website - no Aladin map, etc.

@cdeil

vorugantia committed:
fresh start

vorugantia committed:
update tsconfig.json

vorugantia committed:
update index.html and add load screen

vorugantia committed:
add app.routes.ts (commented out for now)

vorugantia committed:
update

vorugantia committed:
add MapViewComponent and routes to it

vorugantia committed:
add CatViewComponent and route

vorugantia committed:
add SwitchViewComponent (button) that toggles routes /map and /cat

vorugantia committed:
update

vorugantia committed:
fixed unit testing for SwitchViewComponent

vorugantia committed:
add MapComponent

vorugantia committed:
add AboutButtonComponent

vorugantia committed:
add CatSearchComponent

vorugantia committed:
add CatHelpComponent and its route

vorugantia committed:
add CatSource2FHLComponent, its route, fix spec.ts of CatSearchComponent

vorugantia committed:
restructured app + added CatSourceComponent3FGL and its route + dropdown in CatHelpComponent

vorugantia committed:
add CatSourceSNRcatComponent and its route

vorugantia committed:
delete shared files

vorugantia commented:
I'll fix the travis-ci builds first...

vorugantia committed:
comment out spec.ts files

vorugantia commented:
Commented out spec.ts files for now, travis-ci builds passed.

cdeil commented on .gitignore:
```diff
@@ -26,6 +30,4 @@ testem.log
 .DS_Store
-
-.vscode
-__pycache__
```
Don't remove these two lines.

vorugantia replied:
done

vorugantia replied:
done

cdeil commented:
Remove all `.pyc` files, they shouldn't be under version control.
E.g. `gammasky/__pycache__/__init__.cpython-35.pyc`

cdeil commented on package.json:
```diff
@@ -1,13 +1,13 @@
 {
   "name": "gamma-sky",
-  "version": "0.0.1",
+  "version": "0.0.0",
```
Go to 0.0.2 here.

vorugantia replied:
done

cdeil commented on package.json:
```diff
@@ -1,13 +1,13 @@
     "postinstall": "typings install",
     "lint": "tslint \"src/**/*.ts\"",
-    "format": "clang-format -i -style=file --glob=src/**/*.ts",
+    "test": "ng test",
```
Maybe try running ng test also on travis-ci.
If it fails, OK to not run it this way.

cdeil commented on e2e/app.e2e-spec.ts:
```diff
@@ -5,10 +5,10 @@ describe('gamma-sky App', function() {
   it('should display message saying app works', () => {
     page.navigateTo();
-    expect(page.getParagraphText()).toEqual('gamma-sky works!');
+    expect(page.getParagraphText()).toEqual('app works!');
```
Try running `ng e2e` locally. Does it pass? What are these? Should we run them on travis-ci?

vorugantia replied:
Passes locally. I'll add all new travis-ci commands in one commit

cdeil commented on src/app/views/cat-view/cat-view.component.html:
```diff
@@ -0,0 +1,9 @@
+  This is CatViewComponent
+</p>
+
+<switch-view [notCurrentView]="notCurrentView"></switch-view>
```
Rename data member to "selectedView".

vorugantia replied:
done

cdeil commented on src/app/views/switch-view/switch-view.component.html:
```diff
@@ -0,0 +1,5 @@
+<div>
+  <button id='switchViewButton' (click)='onClick($event)'>
+    <b>Go to {{notCurrentView}}</b>
```
if selectedView == 'map':
    "go to cat"
...

vorugantia replied:
done

cdeil commented on src/app/app.routes.ts:
```diff
@@ -0,0 +1,47 @@
+import { CatHelpComponent } from './widgets/cat-help/cat-help.component';
+import { CatSource2FHLComponent } from './widgets/cat-source/cat-source-2fhl/cat-source-2fhl.component';
+import { CatSource3FGLComponent } from './widgets/cat-source/cat-source-3fgl/cat-source-3fgl.component';
+import { CatSourceSNRcatComponent } from './widgets/cat-source/cat-source-snrcat/cat-source-snrcat.component';
```
Could try this here in `src/app/app.routes.ts` (or from anywhere else):

```
import { CatSourceSNRcatComponent } from './widgets/cat-source';
```

And in `app/widgets/cat-source/index.ts` put this:

```
export { CatSourceSNRcatComponent } from './cat-source-snrcat/cat-source-snrcat.component'
```

vorugantia replied:
@cdeil - check out `system-config.ts`, this is where angular-cli generates all barrels. If we want to make our own barrels, I'll have to add them in myself in this file. However, there's a scary "Everything underneath this line is managed by the CLI" message so I'm not sure if I should manually configure barrels.

For this PR, I'm not going to do anything about the barrels and I'll just leave all of the generated `index.ts` files.

cdeil commented:
Add back data files.

Maybe: add back content of about component HTML and CSS

vorugantia committed:
some fixes to PR

vorugantia committed:
add two travis-ci tests

vorugantia committed:
remove $ng test from travis-ci

vorugantia committed:
add SHORTENED (5 sources each) JSON data files

vorugantia committed:
add code for AboutButtonComponent

vorugantia commented:
PR fixes:
- Added data
- Added `ng e2e` to travis-ci (`ng test` didn't work)
- Removed `.pyc` files, fixed .gitignore
- Added code for about button
- gamma-sky v0.0.2

Merging into master...

( @cdeil - see my note in `app.routes.ts` under yours, about barrels.)

cdeil commented:
OK to not introduce other barrels for me. It's not really needed and if it's working now, maybe as you say best just keep as is.

Great progress today, good job!

See you tomorrow.
